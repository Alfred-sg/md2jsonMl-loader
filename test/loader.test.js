import { webpack } from './helpers';

describe('loader', () => {
  it('should works', async () => {
    const stats = await webpack('basic.js');
    const { modules } = stats.toJson();
    const [, test] = modules;

    console.log(test.source);

    expect(test.source).toMatchSnapshot('md');
    expect(stats.compilation.warnings).toMatchSnapshot('warnings');
    // expect(stats.compilation.errors).toMatchSnapshot('errors');
  });
});