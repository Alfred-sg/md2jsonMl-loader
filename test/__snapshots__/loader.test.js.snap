// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loader should works: md 1`] = `"module.exports = {\\"content\\":[\\"article\\",[\\"h1\\",\\"H1\\"],[\\"h2\\",\\"H2\\"],[\\"h3\\",\\"H3\\"],[\\"h4\\",\\"H4\\"],[\\"h5\\",\\"H5\\"],[\\"h6\\",\\"H6\\"],[\\"ol\\",[\\"li\\",[\\"p\\",\\"First\\"]],[\\"li\\",[\\"p\\",\\"Second\\"]],[\\"li\\",[\\"p\\",\\"Third\\"]]],[\\"ul\\",[\\"li\\",[\\"p\\",\\"Something\\"]],[\\"li\\",[\\"p\\",[\\"a\\",{\\"title\\":null,\\"href\\":\\"www.something.com\\"},\\"Something\\"]]],[\\"li\\",[\\"p\\",[\\"img\\",{\\"title\\":null,\\"src\\":\\"www.comething.com/img\\",\\"alt\\":\\"Somethong\\"}]]]],[\\"ol\\",[\\"li\\",[\\"p\\",\\"Parent\\"],[\\"ul\\",[\\"li\\",[\\"p\\",\\"Children\\"]]]]],[\\"table\\",[\\"thead\\",[\\"tr\\",[\\"th\\",\\"hello\\"],[\\"th\\",\\"world\\"]]],[\\"tbody\\",[\\"tr\\",[\\"td\\",\\"你好\\"],[\\"td\\",\\"世界\\"]]]],[\\"p\\",\\"This is a paragraph, including \\",[\\"em\\",\\"EM\\"],\\" and \\",[\\"strong\\",\\"STRONG\\"],\\". Any question? Oh, I almost forget \\",[\\"code\\",\\"inline code\\"],\\".\\"],[\\"pre\\",{\\"lang\\":\\"javascript\\"},[\\"code\\",\\"console.log('Hello world!');\\"]],[\\"blockquote\\",[\\"p\\",\\"Here is a blockquote\\"]],[\\"hr\\"],[\\"script\\",\\"\\\\n  console.log('Good bye all!');\\\\n\\"],[\\"p\\",[\\"a\\",{\\"href\\":\\"benjycui.github.io\\"},\\"My blog\\"]],[\\"div\\",{\\"class\\":\\"home\\"},[\\"div\\",{\\"class\\":\\"banner\\"},\\"Oh, banner!\\"]],[\\"p\\",[\\"br\\"]],[\\"p\\",[\\"span\\",\\"LINK REFERENCE\\"]],[\\"p\\",[\\"img\\",{\\"class\\":\\"preview-img no-padding good\\",\\"align\\":\\"right\\",\\"src\\":\\"https://zos.alipayobjects.com/rmsportal/jeyvhMIQgoPUotNerRGy.png\\",\\"alt\\":\\"正确示例\\"}],\\"\\\\n\\",[\\"img\\",{\\"class\\":\\"preview-img no-padding bad\\",\\"align\\":\\"right\\",\\"src\\":\\"https://zos.alipayobjects.com/rmsportal/ppdlrVnFCsYVicjDrnzi.png\\",\\"alt\\":\\"错误示例\\",\\"description\\":\\"当对比度数值低于 3:1 时，弱视用户将很难识别\\"}]]],\\"meta\\":{\\"title\\":\\"test\\",\\"description\\":\\"a sample Markdown for test\\"}}"`;

exports[`loader should works: warnings 1`] = `Array []`;
